src/com/facebook/buck/features/apple/projectV2/SchemeGenerator.java

```diff
@@ -231,14 +235,20 @@ public Path writeScheme() throws IOException {
     }
 
     ImmutableMap<SchemeActionType, ImmutableMap<String, String>> envVariables = ImmutableMap.of();
+    Map<SchemeActionType, XCScheme.BuildableReference> envVariablesBasedOn = ImmutableMap.of();
     if (environmentVariables.isPresent()) {
       envVariables = environmentVariables.get();
+      if (expandVariablesBasedOn.isPresent()) {
+        envVariablesBasedOn = expandVariablesBasedOn.get().entrySet().stream()
+          .collect(Collectors.toMap(Map.Entry::getKey, e -> buildTargetToBuildableReferenceMap.get(e.getValue())));
+      }
```

Comment:
Similar code is used 3 times. Can it be refactored? Thanks.

Reply:
As I understand `src/com/facebook/buck/features/apple/projectV2` will replace `src/com/facebook/buck/features/apple/project` package in future and there are a lot of duplicates in this packages. I just keep the style of existing classes.